File: Languages/Python/two_sum.py
def two_sum(nums, k):
    if not nums:
        return False

    ht = {}

    for i in range(len(nums)):
        if nums[i] in ht:
            return [ht[nums[i]], i]

        else:
            ht[k - nums[i]] = i
    return False


print(two_sum([10, 15, 3, 7], 17))

File: Languages/Javascript/two-sum/two_sum.js
/**
 *
 * Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
 *
 *  You may assume that each input would have exactly one solution, and you may not use the same element twice.
 *
 *   You can return the answer in any order.
 */

var twoSum = function(nums, target) {
    var result = [];
    var hashTable = {};
      for(i = 0; i < nums.length; i++) {
          let diff = target  - nums[i];
          if((hashTable[diff.toString()]) !== undefined) {
              // console.log(nums[i], diff)
              result.push(i, hashTable[diff.toString()]);
          }
          hashTable[nums[i]] = i;
      }
      
      return result;
  
};